tests/integration-tests/tests/cli_commands/test_cli_commands.py

```diff
@@ -274,9 +274,27 @@ def _test_pcluster_compute_fleet(cluster, expected_num_nodes):
     check_status(cluster, "CREATE_COMPLETE", "running", "RUNNING")
 
 
+def _test_export_log_files_are_expected(cluster, bucket_name, instance_ids, bucket_prefix, filters=None):
+    # test with a prefix and an output file
+    with tempfile.TemporaryDirectory() as tempdir:
+        output_file = f"{tempdir}/testfile.tar.gz"
+        ret = cluster.export_logs(
+            bucket=bucket_name, output_file=output_file, bucket_prefix=bucket_prefix, filters=filters
+        )
+        assert_that(ret["path"]).is_equal_to(output_file)
+
+        with tarfile.open(output_file) as archive:
+            filenames = {logfile.name for logfile in archive}
+
+            # check there are the logs of all the instances and cfn logs
+            for file_expected in set(instance_ids) | {f"{cluster.name}-cfn-events"}:
```

Comment:
With this test we're not sure we have ONLY the logs of the head node.

We could do something like:
```python
if len(set(instance_ids)) == 1:
  # be sure to have only the logs of the specified instance
  assert_that(all(file_expected ...).is_true()
else:
  assert_that(any(file_expected ...).is_true()
```

Reply:
Change to:
```
if len(set(instance_ids)) == 1:  # be sure to have only the logs of the specified instance
    instance_logs = [filename for filename in filenames if ".i-" in filename]
    assert_that(all(instance_ids[0] in instance_log for instance_log in instance_logs)).is_true()
```